packages/client/src/v2-events/features/events/actions/correct/review/Review.tsx

```diff
@@ -0,0 +1,129 @@
+import { FormLayout } from '@client/v2-events/layouts'
+import { ROUTES } from '@client/v2-events/routes'
+import { getScope } from '@client/profile/profileSelectors'
+import { isLastActionCorrectionRequest } from '../utils'
+
+function isRequestCorrectionAction(
+  action: Action
+): action is RequestedCorrectionAction & { declaration: EventState } {
+  return action.type === 'REQUEST_CORRECTION'
+}
+
+export function Review() {
+  const scopes = useSelector(getScope)
+  const { eventId } = useTypedParams(ROUTES.V2.EVENTS.REVIEW_CORRECTION.REVIEW)
+  const events = useEvents()
+
+  const event = events.getEvent.getFromCache(eventId)
+
+  const { eventConfiguration: configuration } = useEventConfiguration(
+    event.type
+  )
+  const eventIndex = getCurrentEventState(event, configuration)
+
+  const formConfig = getDeclaration(configuration)
+
+  const getFormValues = useEventFormData((state) => state.getFormValues)
+  const form = getFormValues()
+
+  const previousFormValues = eventIndex.declaration
+
+  const intlWithData = useIntlFormatMessageWithFlattenedParams()
+
+  const actionConfig = configuration.actions.find(
+    (action) => action.type === ActionType.REQUEST_CORRECTION
+  )
+
+  if (!actionConfig) {
+    throw new Error(
+      `Action config for ${ActionType.REQUEST_CORRECTION} was not found. This should never happen`
+    )
+  }
+
+  const writeActions: ActionDocument[] = getAcceptedActions(event).filter(
+    (a) => !isMetaAction(a.type)
+  )
+  const lastWriteAction = writeActions[writeActions.length - 1]
+
+  const completeDeclaration = getCompleteActionDeclaration(
+    {},
+    event,
+    lastWriteAction
+  )
+  lastWriteAction.declaration = completeDeclaration
+  const lastActionIsCorrectionRequest = isLastActionCorrectionRequest(event)
+
+  const canReviewCorrection =
+    eventIndex.flags.includes(InherentFlags.CORRECTION_REQUESTED) &&
+    scopes?.includes(SCOPES.RECORD_REGISTRATION_CORRECT)
+
+  if (canReviewCorrection && !lastActionIsCorrectionRequest) {
+    throw new Error(
+      `Last action is not a correction request. Last action type: ${lastWriteAction.type}`
+    )
+  }
+
+  const isReviewCorrection =
+    canReviewCorrection && lastActionIsCorrectionRequest
+
+  const formWithNewValues =
+    isReviewCorrection && isRequestCorrectionAction(lastWriteAction)
+      ? deepMerge(form, lastWriteAction.declaration)
+      : form
+
+  return (
+    <FormLayout route={ROUTES.V2.EVENTS.REVIEW_CORRECTION}>
+      <ReviewComponent.Body
+        form={formWithNewValues}
+        formConfig={formConfig}
+        isCorrection={true}
+        isReviewCorrection={isReviewCorrection}
+        previousFormValues={previousFormValues}
+        title={intlWithData.formatMessage(
+          actionConfig.label,
+          previousFormValues
+        )}
+        /*
+         * @todo isReviewCorrection, onEdit & readyonlyMode needs to be reviewed
```

Comment:
worth looking in to right away? would def be nice to find an improvement here :)